proposals/mmmm-spawn-let.md

```diff
@@ -0,0 +1,701 @@
+
+This dataflow pattern from child tasks to parents is very common, and we want to make it as lightweight and safe as possible.
+
+## Proposed solution
+
+This proposal introduces a simple way to create child tasks with and await their results: `spawn let` declarations.
+
+Using `spawn let`, our example looks like this:
+
+```swift
+// given: 
+//   func chopVegetables() async throws -> [Vegetables]
+//   func marinateMeat() async -> Meat
+//   func preheatOven(temperature: Int) -> Oven
+
+func makeDinner() async throws -> Meal {
+  spawn let veggies = chopVegetables()
+  spawn let meat = marinateMeat()
+  spawn let oven = preheatOven(temperature: 350)
+
+  let dish = Dish(ingredients: await [try veggies, meat])
+  return try await oven.cook(dish, duration: .hours(3))
+}
+```
+
+`spawn let` is similar to a `let`, in that it defines a local constant that is initialized by the expression on the right-hand side of the `=`. However, it differs in that the initializer expression is evaluated in a separate, concurrently-executing child task. 
+
+The child task begins running as soon as the `spawn let` is encountered. By default, child tasks use the global, width-limited,  concurrent executor, in the same manner as task group child-tasks do. It is a future direction to allow customizing which executor these should be executing on. On normal completion, the child task will initialize the variables in the `spawn let`.
+
+The right-hand side of a `spawn let` expression can be thought of as an implicit `@Sendable closure`, similar to how the `detach { ... }` API works, however the resulting task is a *child task* of the currently executing task. Because of this, and the need to suspend to await the results of such expression, `spawn let` declarations may only occur within an asynchronous context, i.e. an `async` function or closure.
+
+For single statement expressions in the `spawn let` initializer, the `await` and `try` keywords may be omitted. The effects they represent carry through to the introduced constant and will have to be used on when waiting on the constant. For example, in the example shown above, the veggies are declared as `spawn let veggies = chopVegetables()`, and even through `chopVegetables` is `async` and `throws`, the `await` and `try` keywords do not have to be used on that line of code. Once waiting on the value of that `async let` constant, the compiler will enforce that the expression where the `veggies` appear must be covered by both `await` and some form of `try`.
+
+Because the main body of the function executes concurrently with its child tasks, it is possible that the parent task (the body of `makeDinner` in this example) will reach the point where it needs the value of an `spawn let` (say,`veggies`) before that value has been produced. To account for that, reading a variable defined by an `spawn let` is treated as a potential suspension point,
```

Comment:
```suggestion
Because the main body of the function executes concurrently with its child tasks, it is possible that the parent task (the body of `makeDinner` in this example) will reach the point where it needs the value of a `spawn let` (say,`veggies`) before that value has been produced. To account for that, reading a variable defined by a `spawn let` is treated as a potential suspension point,
```